tkg/tkgctl/helpers.go

```diff
@@ -232,6 +275,13 @@ func (t *tkgctl) processClusterObjectForConfigurationVariables(clusterObj unstru
 	}
 	legacyVarMap[constants.ConfigVariableInfraProvider] = providerName
 
+	// VSPHERE_USERNAME and VSPHERE_PASSWORD do not have mapping to any Cluster variables and should be retrieved from VSphereCluster's identityRef
```

Comment:
Is that possible to add them into cluster variables? Just try to understand which approach is better.

Reply:
We use Secret as vSphereCluster's identityRef which is not a Cluster-API template so Cluster variable can not patch it.